NwInterface.cpp

```diff
@@ -849,14 +849,12 @@ void NwInterface::harvestWatches() {
     trace->traceEvent(TRACE_NORMAL, "last_match=%u / now=%u [to go: %d]",
 		      match->get_last_match(), when, (match->get_last_match() - when));
 #endif
-
-    if(match->ready_to_harvest(when)) {
+    if(match->ready_to_harvest()) {
 #ifdef DEBUG
       trace->traceEvent(TRACE_NORMAL, "Harvesting");
 #endif
       ban((char*)it->first.c_str(), false /* unban */, false, "unban", "");
-      delete it->second;
-      watches_blacklist.erase(it++);    // or "it = m.erase(it)" since C++11
+      it->second->isBanned = false;
```

Comment:
Please describe how harvesting works and justify why the memory does not explode in case of a scan attack or similar.